### PR TITLE
[benchmarks] Fix AMP setup for torchbench models.

### PR DESCRIPTION
**Pull request by ysiraichi**
Fix: #6556 (and, possibly #6833)

This PR fixes the benchmarks script when running with AMP. Previously, we were calling `torch.amp.autocast(..., device_type="xla")` for both XLA:CUDA and XLA:TPU. However, we should be using `torch.cuda.amp.autocast` for XLA:CUDA (see [this](https://github.com/pytorch/xla/blob/1fa1f858342df29ce7cdf2d18cabb259869f652f/torch_xla/csrc/tensor_impl.cpp#L74-L83) for more details).

**Context:** after #6518, `Super_Slomo` inference started being run using AMP. However, due to #6511, that PR tried to mimic `torch_xla.amp.autocast` behavior, using `torch.amp.autocast`.

cc @miladm @JackCaoG @vanbasten23 @zpcore

### CODE REVIEW

**Comment by ysiraichi** [comment]
Confirmed it also fixes #6833.

**Comment by JackCaoG** [comment]
Hmm according to https://github.com/pytorch/xla/blob/master/docs/amp.md we should be able to use `autocast ` for both TPU and GPU, is that no longer the case?

**Comment by ysiraichi** [comment]
That document is correct. Problem is that I didn't notice XLA:CUDA is supposed to run with CUDA autocast, i.e. `torch.amp.autocast("cuda")`. Instead, I was running both XLA:CUDA and XLA:TPU with XLA autocast, i.e. `torch.amp.autocast("xla")`. This behavior is implemented in `torch_xla.amp.autocast`. However, since it currently doesn't work (#6511) with dynamo, I was using `torch.amp.autocast` directly.

**Review by JackCaoG — APPROVED** [review]

File: benchmarks/torchbench_model.py [hunk 1]
```diff
@@ -350,23 +357,27 @@ def conversion_dtype(self):
...
-      if self.benchmark_experiment.accelerator == "cuda":
+      # TODO: Should call device specific autocast implementations.
+      # Specifically, we should be using:
+      #   - torch.cuda.amp.autocast for inductor
+      #   - torch_xla.amp.autocast for PyTorch/XLA experiments.
+      # PyTorch/XLA autocast does not run with dynamo, though:
+      # https://github.com/pytorch/xla/issues/6511
+      if self.is_accelerator_cuda():
+        # For inductor and XLA:CUDA, we use CUDA autocast.
+        autocast = torch.cuda.amp.autocast
```
Comment (vanbasten23) [inline benchmarks/torchbench_model.py:369]: I guess `torch.cuda.amp.autocast` is the same as `torch.amp.autocast("cuda")`?

Comment (vanbasten23) [inline benchmarks/torchbench_model.py:369]: do you need to set `kwargs["device_type"] = "xla"` for XLA:GPU case?

Comment (ysiraichi) [reply benchmarks/torchbench_model.py:369]: Not really. `torch.cuda.amp.autocast` already does that.

**Pull request merged**